src/features/barn/BarnInside.tsx

```diff
@@ -67,13 +70,21 @@ export const BarnInside: React.FC = () => {
 
   mapPlacements.push(...components);
 
+  const nextLevel = Math.min(level + 1, 3) as Exclude<AnimalBuildingLevel, 1>;
+
   return (
     <>
       <AnimalBuildingModal
         buildingName="Barn"
         show={showModal}
         onClose={() => setShowModal(false)}
       />
+      <UpgradeBuildingModal
+        buildingName="Hen House"
+        level={nextLevel}
```

Comment:
I wasn't able to upgrade to level 3 with this configuration. It showed up that I was at max level when I was only level 2. I manually bypassed it to test the final level.